### ISSUE TITLE
Clean up TODOs

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#812]
The following is the complete list of TODOs and will be updated as we resolve them.

TODO list updated on: 2018-02-09, based on the branch `master`, except for the mock module and test module which are based on branch `ks/mock-fix-todos`.

**Discussion Point:** 

* Can we close this issue? At this point, we are done for pywbem 0.12 with this issue about TODOs, because they are all in one of these states:
  - resolved
  - have a separate issue, that is listed in the todo text
  - open for resolution after 0.12 whereby the action we are stating in this issue basically says the same as the todo text. 

************* Module pywbem.cim_http

W:476, 0: TODO #418: Better approach needed than converting to Bytes (fixme)
W:606, 0: TODO #893: Consider the use of default_context() (fixme)

************* Module pywbem.cim_obj

W:710, 0: TODO #1062: Could compare hash values for better performance (fixme)
W:7329, 0: TODO #904: Migrate remaining uses of tocimobj() to cimvalue() and deprecate (fixme)

************* Module pywbem.cim_operations

W:1149, 0: TODO #919: Further tests on this IRETURN or PARAMVALUE (fixme)

************* Module pywbem._listener

W:794, 0: TODO: Describe how the processing threads terminate. (fixme)
* Assign: Andy: Create issue.
* Version: After 0.12
* Status: Open

************* Module pywbem.mof_compiler

W:2118, 0: TODO #991: CreateClass should reject if the class already exists (fixme)
W:2215, 0: TODO #990: Also roll back changes to qualifier declarations (fixme)

************* Module pywbem._server

W:133, 0: TODO: Clarify which OpenPegasus versions need root/PGInterOp? (fixme)
* Assign: Karl: Find out in which version OP introduced support for `interop`.
* Version: After 0.12
* Status: Open

W:693, 0: TODO: Figure out how to get version of SFCB (fixme)
* Assign: Karl: Find an installation with SFCB and provide the property of CIM_ObjectManager that has its version, so we can parse it out.
* Version: After 0.12
* Status: Open

************* Module pywbem.tupleparse

W:1191, 0: TODO #1039: Clarify whether to default omitted qualifier flavors (fixme)
W:1126, 0: TODO #1044: Clarify if hex support is needed. (fixme)
W:1290, 0: TODO #1044: Clarify if hex support is needed. (fixme)
W:1460, 0: TODO #1044: Clarify if hex support is needed (fixme)
W:1490, 0: TODO #1044: Clarify if hex support is needed (fixme)

************* Module pywbem_mock._wbemconnection_mock

W: 78, 0: TODO: ks Future We have not considered that iq and ico are deprecated in (fixme)
W:317, 0: TODO: ks. FUTURE maybe we should really have a subdict per class but (fixme)
W:473, 0: TODO:ks Future we might be able to use MOFWBEMConnection to (fixme)
W:708, 0: TODO:ks FUTURE Consider sorting to perserve order of compile/add. (fixme)
W:728, 0: TODO:ks Future: Possibly sort insts by path order. (fixme)
W:1206, 0: TODO:ks Future. this could become an iterator for efficiency. (fixme)
W:1238, 0: TODO:ks Future Remove dup test since we should be (fixme)
W:1614, 0: TODO:ks FUTURE implement set... method for instance, qualifier, class (fixme)
W:1751, 0: TODO:ks Future use internal function of repo to create namespace (fixme)
W:1980, 0: TODO:ks Future remove this test for duplicate inst paths since (fixme)
W:2245, 0: TODO:ks FUTURE: Make list from _get_reference_classnames if classes (fixme)
W:2549, 0: TODO:ks Future. Use the timeout along with response delay. Then (fixme)

----------------------------------------

************* Module run_cim_operations
W: 85, 0: TODO ks Nov 2016 Create decerators for both this and the OpenPegasus (fixme)
W:225, 0: TODO ks aug 17. This should be integrated into logging (fixme)
W:757, 0: TODO Apparently ks 5/16 Pegasus did not implement all properties (fixme)
W:801, 0: TODO Apparently ks 5/16 Pegasus did not implement all properties (fixme)
W:1203, 0: TODO ks 5/16 why does pegasus return this particular err. (fixme)
W:1382, 0: TODO this only returns single instance. (fixme)
W:1452, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1526, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1575, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1651, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1689, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1721, 0: TODO ks This not right. we are going back to associators (fixme)
W:1763, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1815, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1889, 0: TODO ks 5/30 2016 add tests here (fixme)
W:2361, 0: TODO add get and test for change. (fixme)
W:2531, 0: TODO specalize this one for OpenPegasus (fixme)
W:2545, 0: TODO ks Mar 2017. Returns value 1 rather than zero for some reason (fixme)
W:2551, 0: TODO: Call with empty arrays (fixme)
W:2553, 0: TODO: Call with weird VALUE.REFERENCE child types: (fixme)
W:2580, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:2586, 0: TODO: check return values (fixme)
W:2640, 0: TODO ks dec/16 why returning CIM_Collection (fixme)
W:2641, 0: TODO ks dec/16 why do we not get return when we user resultrole (fixme)
W:2753, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:2898, 0: TODO. This is really broken (fixme)
W:2910, 0: TODO: For now, disabled test for class name of referencing insts. (fixme)
W:3107, 0: TODO add these tests (fixme)
W:3123, 0: TODO validate qualifiers in properties. here or elsewhere (fixme)
W:3128, 0: TODO validate parameters in methods (fixme)
W:3171, 0: TODO could we assert some size limit here. Probably Not (fixme)
W:3310, 0: TODO extend for options of Deepinheritance, LocalOnly, (fixme)
W:3821, 0: TODO account for openpegasus only response of returning (fixme)
W:4026, 0: TODO ks 5/16 figure out why we could not get instances of a class (fixme)
W:4144, 0: TODO create an instance write it, get it and test results (fixme)
W:4199, 0: TODO Create a new instance on server and test return using (fixme)
W:4322, 0: TODO change to create function to do the hiearchy search (fixme)
W:5429, 0: TODO confirm that this should be pegasus (fixme)
W:5772, 0: TODO in the future, return success.  For now because in some (fixme)
W:5894, 0: TODO modify this so it is not Pegasus dependent (fixme)
W:5932, 0: TODO: ks Finish this test completely when we add other changes (fixme)
W:5944, 0: TODO ks 6/16 add more tests including: multiple subscriptions, filters (fixme)
W:6149, 0: TODO Future: modify this so it is not Pegasus dependent (fixme)

************* Module test_cim_obj
W:1045, 0: TODO 01/18 AM Remove test_all() once PR #955 is merged (is in test hash) (fixme)
W:1172, 0: TODO 01/18 AM Reformulate these tests in test hash once PR #955 is merged (fixme)
W:1246, 0: TODO Implement ordering comparison test for CIMInstanceName (fixme)
W:1257, 0: TODO Implement sorting test for CIMInstanceName (fixme)
W:3685, 0: TODO Add tests for CIMInstance property_list argument (fixme)
W:3992, 0: TODO Implement ordering comparison test for CIMInstance (fixme)
W:4003, 0: TODO Implement sorting test for CIMInstance (fixme)
W:6658, 0: TODO Implement ordering comparison test for CIMProperty (fixme)
W:6666, 0: TODO Implement sorting test for CIMProperty (fixme)
W:8393, 0: TODO Implement ordering comparison test for CIMQualifier (fixme)
W:8401, 0: TODO Implement sorting test for CIMQualifier (fixme)
W:8940, 0: TODO 01/18 AM Enable test case once char16 produces single quotes (fixme)
W:11021, 0: TODO Implement ordering comparison test for CIMClass (fixme)
W:11029, 0: TODO Implement sorting test for CIMClass (fixme)
W:12414, 0: TODO Implement ordering comparison test for CIMMethod (fixme)
W:12422, 0: TODO Implement sorting test for CIMMethod (fixme)
W:13988, 0: TODO Implement ordering comparison test for CIMParameter (fixme)
W:13996, 0: TODO Implement sorting test for CIMParameter (fixme)
W:15484, 0: TODO ks Dec 16. Add more equality tests. (fixme)
W:15498, 0: TODO Implement ordering comparison test for CIMQualifierDeclaration (fixme)
W:15506, 0: TODO Implement sorting test for CIMQualifierDeclaration (fixme)
W:15992, 0: TODO Implement tocimxml() test for CIMQualifierDeclaration (fixme)
W:16099, 0: TODO 01/18 AM Enable once char16 uses single quotes (fixme)
W:16446, 0: TODO Implement tocimxml() test for remaining types (fixme)
W:16531, 0: TODO How should pywbem deal with a single, unnamed, keybinding? (fixme)

************* Module test_cim_types
W:405, 0: TODO: Add testcases for get_local_utcoffset() (fixme)
W:406, 0: TODO: Add testcases for now() (fixme)
W:407, 0: TODO: Add testcases for fromtimestamp() (fixme)

************* Module test_cim_xml
W:  8, 0: TODO: A bunch of tests are still unimplemented for bits of the (fixme)
W:490, 0: TODO: Implement ValueNull test (fixme)
W:1255, 0: TODO: (fixme)
W:1342, 0: TODO: RESPONSEDESTINATION (fixme)
W:1375, 0: TODO: RESPONSEDESTINATION (fixme)
W:1485, 0: TODO: QUALIFIER.DECLARATION (fixme)
W:1573, 0: TODO: SIMPLEREQACK (fixme)
W:1699, 0: TODO: INSTANCE* (fixme)
W:1721, 0: TODO: PARAMTYPE (fixme)
W:1781, 0: TODO: QUALIFIER.DECLARATION (fixme)

************* Module test_client
W:805, 0: TODO 2016/07 AM: Improve the presentation of the difference (fixme)
W:852, 0: TODO 2016/07 AM: Improve the presentation of the diff. (fixme)
W:857, 0: TODO redo as indexed loop to compare all items. (fixme)

************* Module test_indicationlistener
W:235, 0: TODO issue 452. Reuse of the indication listener fails at least in py3, (fixme)

************* Module test_itermethods
W: 35, 0: TODO: ks Oct 17 Remove this pylint disable in the future when pylint or (fixme)
W:1634, 0: TODO ks: This assert disabled. (fixme)

************* Module test_logging
W:271, 0: TODO add test (fixme)

************* Module test_mof_compiler
W:328, 0: TODO The following is cim version dependent. (fixme)
W:358, 0: TODO ks 4/16 add error test for something not found (fixme)
W:415, 0: TODO ks 6/16why does this generate end-of-file rather than more (fixme)
W:454, 0: TODO ks apr 2016 Grow the number of functions to test property (fixme)
W:635, 0: TODO add test for array value in inst, not scalar (fixme)
W:664, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:673, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:708, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:716, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)

************* Module test_recorder
W: 88, 0: TODO python 2.7 will not write the following unicode character (fixme)
W:192, 0: TODO add test for reference properties (fixme)
W:193, 0: TODO add test for embedded object property (fixme)
W:252, 0: TODO host does not appear in output yaml. Is that correct??? (fixme)
W:750, 0: TODO. V2 valid string has extra single quote after CIMError (fixme)

************* Module test_tupleparse
W:6863, 0: TODO 1/18 AM: Should this not be rejected as invalid type? (fixme)
W:9588, 0: TODO: IPARAMVALUE tests (fixme)
W:9589, 0: TODO: PARAMVALUE tests (fixme)
W:9590, 0: TODO: EXPPARAMVALUE tests (fixme)
W:9591, 0: TODO: RETURNVALUE tests (fixme)
W:9592, 0: TODO: IRETURNVALUE tests (fixme)
W:9593, 0: TODO: IMETHODCALL tests (fixme)
W:9594, 0: TODO: IMETHODRESPONSE tests (fixme)
W:9595, 0: TODO: METHODCALL tests (fixme)
W:9596, 0: TODO: METHODRESPONSE tests (fixme)
W:9597, 0: TODO: EXPMETHODCALL tests (fixme)
W:9598, 0: TODO: EXPMETHODRESPONSE tests (fixme)
W:9599, 0: TODO: MESSAGE tests (fixme)
W:9600, 0: TODO: SIMPLEREQ tests (fixme)
W:9601, 0: TODO: SIMPLERSP tests (fixme)
W:9602, 0: TODO: SIMPLEEXPREQ tests (fixme)
W:9603, 0: TODO: SIMPLEEXPRSP tests (fixme)

************* Module test_tupletree
W:166, 0: TODO: Cannot handle comments, cdata, processing instructions, etc. (fixme)

************* Module test_wbemconnection_mock
W:892, 0: TODO expand to use lo parameterized.  Note that (fixme)
W:1019, 0: TODO: Add test of format of the outut with a very simple schema to keep (fixme)
W:1181, 0: TODO modify the merge function to clear each namespace in the repo. (fixme)
W:1325, 0: TODO test returned instance (fixme)
W:1595, 0: TODO add detailed test for di (fixme)
W:1645, 0: TODO implement this one. (fixme)
W:1656, 0: TODO Add tests test_enumerateclasses lo (fixme)
W:1657, 0: TODO Add more precise test for class where we get back original from (fixme)
W:1781, 0: TODO test returned instance. (fixme)
W:1806, 0: TODO extend to test lo and iq (fixme)
W:1817, 0: TODO add tests for iq and ico (fixme)
W:1836, 0: TODO extend to test lo and iq (fixme)
W:1847, 0: TODO test complete instances (fixme)
W:1957, 0: TODO integrate this into normal test (fixme)
W:1992, 0: TODO change to use algorithm to get list of possible (fixme)
W:2018, 0: TODO compute number should be returned. This is number with (fixme)
W:2061, 0: TODO compute number should be returned. (fixme)
W:2078, 0: TODO improve algorithm so we define exactly what properties (fixme)
W:2098, 0: TODO Test actual instance returned (fixme)
W:2139, 0: TODO Test actual instance returned (fixme)
W:2147, 0: TODO repeat pl with conn (fixme)
W:2171, 0: TODO Test actual instance returned (fixme)
W:2208, 0: TODO: Still have issue with one test. (fixme)
W:2221, 0: TODO Test actual instance returned (fixme)
W:2223, 0: TODO test for ico and iq (fixme)
W:2224, 0: TODO test for pl where repository contains classes (fixme)
W:2225, 0: TODO test for instances do not exist. (fixme)
W:2228, 0: TODO integrate this into normal test (fixme)
W:2449, 0: TODO additional tests. (fixme)
W:2723, 0: TODO fix common test for path equalityassert paths == exp_total_paths (fixme)
W:2744, 0: TODO test actual instances (fixme)
W:2966, 0: TODO test instances returned rather than just paths and (fixme)
W:2995, 0: TODO FUTURE: test with timer (fixme)
W:3335, 0: TODO expand test exp_rslt definition. Need to be able to (fixme)
W:3353, 0: TODO enable the above tests (fixme)
W:3379, 0: TODO redo this as a clean compare test (fixme)
W:3396, 0: TODO not sure we really need this testcase? Combine with next case (fixme)
W:3416, 0: TODO test for specific instance. (fixme)
W:3489, 0: TODO test for references propertylist. (fixme)
W:3505, 0: TODO I don't think this one is correct (fixme)
W:3583, 0: TODO add more tests (fixme)
W:3630, 0: TODO expand associator instance names tests (fixme)
W:3657, 0: TODO add more tests (fixme)
W:3706, 0: TODO expand associator instance tests (fixme)
W:3716, 0: TODO I don't think this one is correct (fixme)
W:3737, 0: TODO test error.  Failed if rc was TST_PERSON (fixme)
W:3771, 0: TODO expand associator classes test to test for correct properties (fixme)
W:3838, 0: TODO extend so generates whatever exception defined (fixme)

**Comment by andy-maier** [comment]
New list of TODOs as of 2017-12-14, with PR #865:

************* Module pywbem._server
W:140, 0: TODO: Clarify which OpenPegasus versions need root/PGInterOp? (fixme)
W:700, 0: TODO: Figure out how to get version of SFCB (fixme)
W:962, 0: TODO: Change to min(cimtype) once issue #268 is solved. (fixme)
W:973, 0: TODO: Change to max(cimtype) once issue #268 is solved. (fixme)
W:1013, 0: TODO: We should make type_from_name() and cimtype() part of the API (fixme)
************* Module pywbem.mof_compiler
W:804, 0: TODO flavor trash. (fixme)
W:817, 0: TODO flavor trash. (fixme)
W:968, 0: TODO propagated? (fixme)
W:991, 0: TODO 8/16 ks Consider complete removed of TOINSTANCE from compiler (fixme)
W:1118, 0: TODO what to do with propagated? (fixme)
W:1357, 0: TODO deal with non-decimal values. (fixme)
W:2097, 0: TODO 2016/03 AM: Dubious stmt above. (fixme)
W:2118, 0: TODO: should we see if it exists first with (fixme)
W:2217, 0: TODO: We want rollback to do something with qualifiers? (fixme)
************* Module pywbem.tupleparse
W: 70, 0: TODO: Maybe take a DTD fragment like "(DECLGROUP | (fixme)
W: 74, 0: TODO: Syntax-check some attributes with defined formats, such as NAME (fixme)
W: 76, 0: TODO: Implement qualifiers by making subclasses of CIM types with a (fixme)
W:758, 0: FIXME: This is probably not the best representation of these forms... (fixme)
W:829, 0: XXX: Use TYPE attribute to create named CIM type. (fixme)
W:833, 0: XXX: Would like to use long() here, but that tends to cause (fixme)
W:869, 0: TODO: Return these as maps, not lists (fixme)
W:903, 0: XXX: This does not enforce ordering constraint (fixme)
W:905, 0: XXX: This does not enforce the constraint that there be only (fixme)
W:908, 0: TODO: Parse instance qualifiers (fixme)
W:1021, 0: TODO: Lift this out? (fixme)
W:1054, 0: TODO: Parse this into NAME, VALUE, where the value contains (fixme)
W:1135, 0: TODO: qualifiers, other attributes (fixme)
W:1457, 0: TODO ks 6/16 Extended per DSP0201 v 1.4 to include CLASSNAME, (fixme)
W:1605, 0: TODO ks 5/16this would be more effective if we had a function that (fixme)
W:1624, 0: TODO: Return a CIMError object, not a tuple (fixme)
W:1677, 0: XXX: doesn't prohibit the case of only one VALUE.ARRAY or (fixme)
W:1691, 0: TODO: Call unpack_value if appropriate (fixme)
W:1769, 0: TODO ks 11/14/16 Does not handle exceptions.  See cim_operations l 800 (fixme)
W:1785, 0: TODO: Handle VALUE.REFERENCE, VALUE.REFARRAY (fixme)
************* Module pywbem.cim_obj
W:700, 0: TODO does not account for the extra char that may be appended (fixme)
W:1378, 0: TODO doesn't handle double-quoting, as in refs to refs.  Example: (fixme)
W:1763, 0: TODO: Clarify why we ignore such properties. (fixme)
W:3355, 0: TODO ks 8/16 do special formatting for this so output (fixme)
W:3852, 0: TODO is None allowed for return type. (fixme)
W:3856, 0: TODO CIM-XML does not support methods returning reference (fixme)
W:5390, 0: TODO: Verify whether this is a bug to have this test after the one for (fixme)
W:5450, 0: TODO: Move remaining internal uses of tocimobj() to cimvalue() and deprecate (fixme)
W:5580, 0: TODO doesn't handle double-quoting, as in refs to refs.  Example: (fixme)
************* Module pywbem._listener
W:825, 0: TODO: Describe how the processing threads terminate. (fixme)
************* Module test_logging
W:271, 0: TODO add test (fixme)
************* Module run_operationtimeouts
************* Module test_tupletree
W:165, 0: XXX: Cannot handle comments, cdata, processing instructions, etc. (fixme)
************* Module test_cim_xml
W:  8, 0: TODO: A bunch of tests are still unimplemented for bits of the (fixme)
W:488, 0: TODO: Implement ValueNull test (fixme)
W:1253, 0: TODO: (fixme)
W:1340, 0: TODO: RESPONSEDESTINATION (fixme)
W:1373, 0: TODO: RESPONSEDESTINATION (fixme)
W:1483, 0: TODO: QUALIFIER.DECLARATION (fixme)
W:1571, 0: TODO: SIMPLEREQACK (fixme)
W:1697, 0: TODO: INSTANCE* (fixme)
W:1719, 0: TODO: PARAMTYPE (fixme)
W:1779, 0: TODO: QUALIFIER.DECLARATION (fixme)
************* Module pywbem.cim_http
W:126, 0: TODO 5/16 ks This is a linux based set of defaults: (fixme)
W:482, 0: TODO AM: Should we set strict=True in the following call, for PY2? (fixme)
W:491, 0: TODO AM: Should we set strict=True in the following call, for PY2? (fixme)
W:577, 0: TODO 2/16 AM: Verify whether the additional exceptions (fixme)
W:590, 0: TODO ks 4/16: confirm that we cannot use the default_context() (fixme)
W:595, 0: TODO ks 4/16: Is there a use for the CERT_OPTIONAL mode (fixme)
W:721, 0: TODO 3/16 AM: Track resolution of this Pylint bug. (fixme)
************* Module test_cim_operations
************* Module test_indicationlistener
W:235, 0: TODO issue 452. Reuse of the indication listener fails at least in py3, (fixme)
************* Module pywbem.cim_operations
W:1239, 0: TODO 3/16 AM: Clean up exception handling. The next two lines are a (fixme)
W:1342, 0: TODO Further tests on this IRETURN or PARAMVALUE (fixme)
W:1483, 0: TODO 3/16 AM: Clean up exception handling. The next two lines are a (fixme)
W:1924, 0: TODO ks 6/16 would the setattr be faster? (fixme)
W:1925, 0: TODO: ks 6/16 should we check before setting? (fixme)
W:6635, 0: TODO: Why not accept returned namespace? (fixme)
************* Module test_nocasedict
W:283, 0: TODO: swapcase2() is also defined in test_cim_obj.py. Consolidate. (fixme)
************* Module test_tupleparse
W:112, 0: TODO 2/16 KS: Create test for more complete class. (fixme)
W:170, 0: TODO 2/16 KS: Extend to all property data types. (fixme)
W:208, 0: TODO 2/16 KS: Extend for all data types. (fixme)
************* Module test_client
W:801, 0: TODO 2016/07 AM: Improve the presentation of the difference (fixme)
W:848, 0: TODO 2016/07 AM: Improve the presentation of the diff. (fixme)
W:853, 0: TODO redo as indexed loop to compare all items. (fixme)
************* Module test_itermethods
W: 35, 0: TODO: ks Oct 17 Remove this pylint disable in the future when pylint or (fixme)
************* Module test_cim_types
W:400, 0: TODO: Add testcases for get_local_utcoffset() (fixme)
W:401, 0: TODO: Add testcases for now() (fixme)
W:402, 0: TODO: Add testcases for fromtimestamp() (fixme)
************* Module test_mof_compiler
W:297, 0: TODO The following is cim version dependent. (fixme)
W:327, 0: TODO ks 4/16 add error test for something not found (fixme)
W:384, 0: TODO ks 6/16why does this generate end-of-file rather than more (fixme)
W:423, 0: TODO ks apr 2016 Grow the number of functions to test property (fixme)
W:575, 0: TODO add test for array value in inst, not scalar (fixme)
W:604, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:613, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:644, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:652, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:1257, 0: TODO: The current floatValue regexp does not match, so (fixme)
W:1260, 0: TODO: This testcase succeeds without any expected token for the (fixme)
************* Module test_recorder
W: 84, 0: TODO python 2.7 will not write the following unicode character (fixme)
W:188, 0: TODO add test for reference properties (fixme)
W:189, 0: TODO add test for embedded object property (fixme)
W:246, 0: TODO host does not appear in output yaml. Is that correct??? (fixme)
W:286, 0: TODO the following fails because of issues with timedelta (fixme)
W:307, 0: TODO test result  This test goes away when we fix timedelta yaml (fixme)
W:715, 0: TODO. V2 valid string has extra single quote after CIMError (fixme)
************* Module run_cim_operations
W: 85, 0: TODO ks Nov 2016 Create decerators for both this and the OpenPegasus (fixme)
W:225, 0: TODO ks aug 17. This should be integrated into logging (fixme)
W:757, 0: TODO Apparently ks 5/16 Pegasus did not implement all properties (fixme)
W:801, 0: TODO Apparently ks 5/16 Pegasus did not implement all properties (fixme)
W:1203, 0: TODO ks 5/16 why does pegasus return this particular err. (fixme)
W:1382, 0: TODO this only returns single instance. (fixme)
W:1452, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1526, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1575, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1651, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1689, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1721, 0: TODO ks This not right. we are going back to associators (fixme)
W:1763, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1815, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1889, 0: TODO ks 5/30 2016 add tests here (fixme)
W:2353, 0: TODO add get and test for change. (fixme)
W:2523, 0: TODO specalize this one for OpenPegasus (fixme)
W:2537, 0: TODO ks Mar 2017. Returns value 1 rather than zero for some reason (fixme)
W:2543, 0: TODO: Call with empty arrays (fixme)
W:2545, 0: TODO: Call with weird VALUE.REFERENCE child types: (fixme)
W:2572, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:2578, 0: TODO: check return values (fixme)
W:2632, 0: TODO ks dec/16 why returning CIM_Collection (fixme)
W:2633, 0: TODO ks dec/16 why do we not get return when we user resultrole (fixme)
W:2745, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:2890, 0: TODO. This is really broken (fixme)
W:2902, 0: TODO: For now, disabled test for class name of referencing insts. (fixme)
W:3099, 0: TODO add these tests (fixme)
W:3115, 0: TODO validate qualifiers in properties. here or elsewhere (fixme)
W:3120, 0: TODO validate parameters in methods (fixme)
W:3163, 0: TODO could we assert some size limit here. Probably Not (fixme)
W:3302, 0: TODO extend for options of Deepinheritance, LocalOnly, (fixme)
W:3804, 0: TODO account for openpegasus only response of returning (fixme)
W:4009, 0: TODO ks 5/16 figure out why we could not get instances of a class (fixme)
W:4127, 0: TODO create an instance write it, get it and test results (fixme)
W:4182, 0: TODO Create a new instance on server and test return using (fixme)
W:4305, 0: TODO change to create function to do the hiearchy search (fixme)
W:5412, 0: TODO confirm that this should be pegasus (fixme)
W:5755, 0: TODO in the future, return success.  For now because in some (fixme)
W:5877, 0: TODO modify this so it is not Pegasus dependent (fixme)
W:5915, 0: TODO: ks Finish this test completely when we add other changes (fixme)
W:5927, 0: TODO ks 6/16 add more tests including: multiple subscriptions, filters (fixme)
W:6132, 0: TODO Future: modify this so it is not Pegasus dependent (fixme)
************* Module test_cim_obj
W:848, 0: TODO: Verify DeprecationWarning after migrating to py.test (fixme)
W:1046, 0: TODO Implement ordering comparison test for CIMInstanceName (fixme)
W:1057, 0: TODO Implement sorting test for CIMInstanceName (fixme)
W:1255, 0: TODO: Improve implementation to make it fail for the disabled cases. (fixme)
W:1797, 0: TODO Add tests for CIMInstance property_list argument (fixme)
W:2040, 0: TODO Implement ordering comparison test for CIMInstance (fixme)
W:2051, 0: TODO Implement sorting test for CIMInstance (fixme)
W:2323, 0: TODO this test only catchs existence of the embedded instance (fixme)
W:3290, 0: TODO Implement ordering comparison test for CIMProperty (fixme)
W:3298, 0: TODO Implement sorting test for CIMProperty (fixme)
W:3575, 0: TODO Implement ordering comparison test for CIMQualifier (fixme)
W:3583, 0: TODO Implement sorting test for CIMQualifier (fixme)
W:3893, 0: TODO: Improve implementation to make it fail for the disabled cases. (fixme)
W:4299, 0: TODO Implement ordering comparison test for CIMClass (fixme)
W:4307, 0: TODO Implement sorting test for CIMClass (fixme)
W:4504, 0: TODO ks apr 16: extend this test for other alternatives for mof output (fixme)
W:4822, 0: TODO Implement ordering comparison test for CIMMethod (fixme)
W:4830, 0: TODO Implement sorting test for CIMMethod (fixme)
W:5231, 0: TODO Implement ordering comparison test for CIMParameter (fixme)
W:5239, 0: TODO Implement sorting test for CIMParameter (fixme)
W:5565, 0: TODO ks Dec 16. Add more equality tests. (fixme)
W:5579, 0: TODO Implement ordering comparison test for CIMQualifierDeclaration (fixme)
W:5587, 0: TODO Implement sorting test for CIMQualifierDeclaration (fixme)
W:5643, 0: TODO Implement tocimxml() test for CIMQualifierDeclaration (fixme)
W:5651, 0: TODO Implement tocimxml() test for remaining types (fixme)
W:5735, 0: TODO How should pywbem deal with a single, unnamed, keybinding? (fixme)
W:6126, 0: TODO: This may be wrong in that it breaks a word, not between words. (fixme)
